shared/actions/fs/index.js

```diff
@@ -516,13 +523,18 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
       throw err
     }
   }
-  throw new Error('failed to load mime type')
+  throw new Error('exceeded max retries')
 }
 
-const loadMimeType = (action: FsGen.MimeTypeLoadPayload) =>
-  Saga.call(_loadMimeType, action.payload.path, action.payload.refreshTag)
+function* loadMimeType(action: FsGen.MimeTypeLoadPayload) {
+  try {
+    yield Saga.call(_loadMimeType, action.payload.path, action.payload.refreshTag)
+  } catch (error) {
+    yield Saga.put(makeUnretriableErrorHandler(action)(error))
+  }
+}
 
-const commitEdit = (action: FsGen.CommitEditPayload, state: TypedState) => {
+const commitEdit = (state: TypedState, action: FsGen.CommitEditPayload) => {
```

Comment:
Since we always return a promise now, you can type the return type here.

Reply:
again, not really needed. flow will tell you when it needs to be typed

Reply:
This is used in `actionToPromise` below, so it's not vague to humans. @jzila I'm gonna keep it this way unless you have a strong opinion here.